include/odp/api/spec/dma_types.h

```diff
@@ -404,6 +411,36 @@ typedef struct odp_dma_seg_t {
 		 *  with other data formats. */
 		odp_packet_t packet;
 
+		/** Packet index configuration */
+		struct {
+			/** Packet index
+			 *
+			 *  When odp_dma_transfer_param_t::seg_alloc is used, implementation
+			 *  allocates all required destination packets and this index field can be
+			 *  used to define which packet of the to-be-allocated array of packets is
+			 *  used as the destination for this particular destination segment
+			 *  descriptor. Same index can be reused across subsequent segments if
+			 *  application wishes to transfer to different offsets of the same
+			 *  destination packet. This means that if application for example uses
+			 *  segment index 0 in each destination segment descriptor (with different
+			 *  'offset' values), implementation would allocate only a single packet
+			 *  with data being transferred to different offsets of that single
+			 *  destination packet. Indexes should appear in ascending order, starting
+			 *  from index 0.
+			 */
+			uint16_t index;
```

Comment:
For consistency (to match pkt_len), name of this field could be pkt_index. Also, "pkt_index" is less likely to collide later if another "index" field needs to be added into this struct.

Reply:
I believe the naming is now fine, but additionally the type of the index could be changed from `uint16_t` to `uint32_t`, this would then be more consistent with `odp_dma_transfer_param_t::num_dst` and `odp_dma_result_t::num_dst` which are both `uint32_t`.

Reply:
Thank you for fixing the field name. I agree with Tuomas that uint32_t could be used instead. It does not increase struct size and harmonizes numbers DMA segments/packets parameter types.